addons/mail/models/ir_attachment.py

```diff
@@ -71,5 +82,98 @@ def _attachment_format(self, commands=False):
                     'res_id': attachment.res_id,
                     'res_model': attachment.res_model,
                 })
-            res_list.append(res)
-        return res_list
+            attachments.append(res)
+        return attachments
+
+    @api.model
+    def _get_data_from_url(self, url):
+        """
+        This will create attachment data based on what we can read from the URL.
+        If the URL is an HTML page, this will parse the OpenGraph meta data.
+        If the URL is an image, this will create an image attachment.
+        """
+        try:
+            page = requests.get(url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
```

Comment:
I m not sure this is desirable. This would accept everything between 200 and 400 code, but this code will not know what to do with a 306 response.
I m also not sure what would happen if we hit 307.
I think limiting this to 200 is safer.

Reply:
Yeah I don't really see how we would handle codes other than 200. Also I would hope that the various redirect codes are automatically followed internally?